psi4/src/psi4/libfunctional/LibXCfunctional.cc

```diff
@@ -227,50 +227,60 @@ void LibXCFunctional::set_tweak(std::vector<double> values) {
     if (xc_func_name_ == "XC_GGA_X_B86") {
         if (vsize == 3) {
             // (XC(func_type) *p, FLOAT beta, FLOAT gamma, FLOAT omega);
-            xc_gga_x_b86_set_params(xc_functional_.get(), values[0], values[1], values[2]);
+            // xc_gga_x_b86_set_params(xc_functional_.get(), values[0], values[1], values[2]);
+            xc_func_set_ext_params(xc_functional_.get(), values.data());
             failed = false;
         }
     } else if (xc_func_name_ == "XC_GGA_X_B88") {
         if (vsize == 2) {
             // (XC(func_type) *p, FLOAT beta, FLOAT gamma);
-            xc_gga_x_b88_set_params(xc_functional_.get(), values[0], values[1]);
+            // xc_gga_x_b88_set_params(xc_functional_.get(), values[0], values[1]);
+            xc_func_set_ext_params(xc_functional_.get(), values.data());
             failed = false;
         }
     } else if (xc_func_name_ == "XC_GGA_X_PBE") {
         if (vsize == 2) {
-            //  (XC(func_type) *p, FLOAT kappa, FLOAT mu);
-            xc_gga_x_pbe_set_params(xc_functional_.get(), values[0], values[1]);
+            // (XC(func_type) *p, FLOAT kappa, FLOAT mu);
+            // xc_gga_x_pbe_set_params(xc_functional_.get(), values[0], values[1]);
+            xc_func_set_ext_params(xc_functional_.get(), values.data());
             failed = false;
         }
     } else if (xc_func_name_ == "XC_GGA_C_PBE") {
-        if (vsize == 1) {
-            // (XC(func_type) *p, FLOAT beta);
-            xc_gga_c_pbe_set_params(xc_functional_.get(), values[0]);
+        if (vsize == 3) {
+            // (XC(func_type) *p, FLOAT beta); FLOAT gamma, FLOAT BB
+            // xc_gga_c_pbe_set_params(xc_functional_.get(), values[0]);
+            values[1] = xc_func_info_get_ext_params_default_value(const_cast<xc_func_info_type*>(xc_functional_->info), 1);
+            values[2] = xc_func_info_get_ext_params_default_value(const_cast<xc_func_info_type*>(xc_functional_->info), 2);
+            xc_func_set_ext_params(xc_functional_.get(), values.data());
             failed = false;
         }
     } else if (xc_func_name_ == "XC_GGA_X_PW91") {
         if (vsize == 7) {
             // (XC(func_type) *p, FLOAT a, FLOAT b, FLOAT c, FLOAT d, FLOAT f, FLOAT alpha, FLOAT expo);
-            xc_gga_x_pw91_set_params(xc_functional_.get(), values[0], values[1], values[2], values[3], values[4], values[5],
-                                     values[6]);
+            // xc_gga_x_pw91_set_params(xc_functional_.get(), values[0], values[1], values[2], values[3], values[4], values[5],
+            //                          values[6]);
+            //xc_func_set_ext_params(xc_functional_.get(), values.data());
```

Comment:
It looks like they simplified their PW91 interface:
```
static const func_params_type ext_params[] = {
  {"_bt",    0.00426, "a = 6 bt/X2S"},
  {"_alpha", 100.0,   "parameter of the exponential term"},
  {"_expo",  3.72,    "exponent of the power in the numerator"},
};

static void
set_ext_params(xc_func_type *p, const double *ext_params)
{
  gga_x_pw91_params *params;
  double bt, beta;

  assert(p != NULL && p->params != NULL);
  params = (gga_x_pw91_params *) (p->params);

  bt = get_ext_param(p->info->ext_params, ext_params, 0);
  params->alpha = get_ext_param(p->info->ext_params, ext_params, 1);
  params->expo  = get_ext_param(p->info->ext_params, ext_params, 2);


  beta         =  5.0*pow(36.0*M_PI,-5.0/3.0);
  params->a    =  6.0*bt/X2S;
  params->b    =  1.0/X2S;
  params->c    =  bt/(X_FACTOR_C*X2S*X2S);
  params->d    = -(bt - beta)/(X_FACTOR_C*X2S*X2S);
  params->f    =  1.0e-6/(X_FACTOR_C*pow(X2S, params->expo));
}
```

This should make our `hyb_funcs.py:def get_pw6b95_tweaks():` simpler. It should not be seg faulting however, the values should be as shown in the functions:
```
    bt = 0.004440  # paper values
    c_pw = 0.32620  # paper values
    expo_pw6 = 3.7868  # paper values
    params = [bt, c_pw / X2S / X2S, expo_pw6]
```

I believe

Reply:
I tried some 3-term pw91 to poor effect. But I suspect I'm still at fault.